java/rocksjni/parsed_entry_info.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// This file implements the "bridge" between Java and C++ and enables
+// calling c++ ROCKSDB_NAMESPACE::Iterator methods from Java side.
+
+#include <jni.h>
+#include <stdlib.h>
+
+#include "include/org_rocksdb_ParsedEntryInfo.h"
+#include "rocksdb/options.h"
+#include "rocksdb/types.h"
+#include "rocksdb/utilities/types_util.h"
+#include "rocksjni/cplusplus_to_java_convert.h"
+#include "rocksjni/portal.h"
+
+/*
+ * Class:     org_rocksdb_ParsedEntryInfo
+ * Method:    newParseEntryInstance
+ * Signature: ()J
+ */
+jlong JNICALL Java_org_rocksdb_ParsedEntryInfo_newParseEntryInstance(
+    JNIEnv * /*env*/, jclass /*cls*/) {
+  ROCKSDB_NAMESPACE::ParsedEntryInfo *parsed_entry_info =
+      new ROCKSDB_NAMESPACE::ParsedEntryInfo();
+  return GET_CPLUSPLUS_POINTER(parsed_entry_info);
+}
+
+/*
+ * Class:     org_rocksdb_ParsedEntryInfo
+ * Method:    parseEntry
+ * Signature: (JJ[BI)V
+ */
+void JNICALL Java_org_rocksdb_ParsedEntryInfo_parseEntry(
+    JNIEnv *env, jclass /*cls*/, jlong handle, jlong options_handle,
+    jbyteArray jtarget, jint len) {
+  auto *options =
+      reinterpret_cast<const ROCKSDB_NAMESPACE::Options *>(options_handle);
+  auto *parsed_entry_info =
+      reinterpret_cast<ROCKSDB_NAMESPACE::ParsedEntryInfo *>(handle);
+  jbyte *target = env->GetByteArrayElements(jtarget, nullptr);
+  if (target == nullptr) {
+    jclass oom_class = env->FindClass("java/lang/OutOfMemoryError");
```

Comment:
There is a `portal.h` utility method for this - `OutOfMemoryErrorJni::ThrowNew(env, msg)` please use it. (and in the methods below)

Reply:
done